pkg/flow/temporary.go

```diff
@@ -238,13 +238,17 @@ func (im *instanceMemory) SetVariables(ctx context.Context, vars []states.Variab
 			return derrors.NewInternalError(errors.New("invalid scope"))
 		}
 
-		_, _, err = im.engine.flow.SetVariable(ctx, vrefc, vdatac, q, v.Key, v.Data, v.MIMEType, thread)
-		if err != nil {
-			return err
-		}
-
-		if err != nil {
-			return err
+		d := string(v.Data)
+		if len(d) < 3 { // checking for < 3 cause empty data looks like {}
```

Comment:
I think this has some edge-case bugs. If you look at `pkg/flow/states/setter.go` you'll find the only place that calls this function. Immediately above the call to this function, you'll see that sometimes the output might not be JSON at all. Furthermore, there is valid non-empty JSON data that is shorter than 3 characters. For example: `5` is perfectly valid JSON on its own. 

We probably need to be a bit more cautious about what constitutes a null value. If there's zero bytes of data, that's zero. 

IF the data is being treated as JSON: 
- If it's `{}`, that's null. 
- If it's `[]`, that's null. 
- If it's `0`, that's null. 
- If it's `""`, that's null. 
- And if it's `null`, that's also null. 

But be extra careful that it is indeed being treated as JSON. Look at the if/else-if/else beginning on `pkg/flow/states/setter.go:98`. It's only being treated as JSON if it gets to the "else" branch.